src/internal/client/client.go

```diff
@@ -585,8 +585,11 @@ func newOnUserMachine(ctx context.Context, cfg *config.Config, context *config.C
 		client.authenticationToken = context.SessionToken
 	}
 
-	// Verify cluster deployment ID
+	// Verify cluster deployment ID and project.
 	clusterInfo, err := client.InspectClusterWithVersionAndProject(version.Version, &pfs.Project{Name: context.Project})
+	if err != nil && status.Code(err) == codes.Unauthenticated {
```

Comment:
Should we also check for `codes.ErrBadToken`? In that case, the use would also need to login.

Reply:
There is no `codes.ErrBadToken`; I think that you mean `auth.ErrBadToken`, which is a `codes.Unauthenticated` error: https://github.com/pachyderm/pachyderm/blob/325662980c95803503c419c0c94f100bdd95182d/src/auth/auth.go#L141

Arguably we should also retry when the error is `codes.PermissionDenied`, though. I’ll push that update in a second.